### ISSUE TITLE
Sentry is not reporting line numbers correctly

### ISSUE THREAD

**Issue by timwright12** [department-of-veterans-affairs/va.gov-team#39488]
## Description

When we have errors reporting in Sentry they are supposed to display the line number associated with each error to help with debugging. Currently it isn't reporting a helpful value and looks like the sourcemap isn't being generated correctly.

We're seeing output like: `Error in yu at line 2:430148`, which is indicative (to me) of a sourcemap being generated from minified & obfuscated code rather than sourcecode

### Resources

- [4 Reasons Why Your Source Maps are Broken](https://blog.sentry.io/2018/10/18/4-reasons-why-your-source-maps-are-broken)

## Examples:

[TypeError ?(generated/claims-status.entry)](http://sentry.vfs.va.gov/organizations/vsp/issues/92395/?project=4&query=is%3Aunresolved&statsPeriod=14d)

## Definition of done

- [ ] Sourcemaps are generating correctly
- [ ] Line numbers are being reported correctly in Sentry

**Comment by acrollet** [comment]
Some notes from my investigation:

- Sentry has a source map tool available at https://sourcemaps.io/
- It is happy with some source maps on va.gov: https://sourcemaps.io/report/1648847132206_https%3A%2F%2Fwww.va.gov%2Fgenerated%2F670.entry.js
- It fails to validate [check-in.entry.js](https://prod-va-gov-assets.s3-us-gov-west-1.amazonaws.com/generated/check-in.entry.js): https://sourcemaps.io/report/1648846994669_https%3A%2F%2Fprod-va-gov-assets.s3-us-gov-west-1.amazonaws.com%2Fgenerated%2Fcheck-in.entry.js
- I noticed that `check-in.entry.js` is served with gzip encoding and `670.entry.js` is not
- Sentry says that [it cannot deal with gzipped artifacts](https://docs.sentry.io/platforms/javascript/sourcemaps/troubleshooting_js/#verify-artifacts-are-not-gzipped)

**Comment by acrollet** [comment]
One other note, sentry says that [a release must be created in sentry](https://docs.sentry.io/platforms/javascript/sourcemaps/troubleshooting_js/#verify-a-release-is-configured-in-your-sdk) for uploaded artifacts to be applied. I'm thinking/hoping that won't matter for automatically fetched source maps but I thought I'd mention it since it doesn't look like we have releases configured for vets-website in sentry.

**Comment by timwright12** [comment]
I reworked how we're generating sourcemaps in vets-website to see if that was the issue. I also updated the allowed_origins list in devops just incase it's producing a CORS error when requesting the map. If none of that works, we can always try uploading sourcemaps directly to Sentry in CI

**Comment by timwright12** [comment]
Sourcemaps seem to be validating, but still not reporting line numbers: https://sourcemaps.io/report/1649190020681_https%3A%2F%2Fwww.va.gov%2Fgenerated%2F651.entry.js

Chasing down the CORS theory right now as well

**Comment by acrollet** [comment]
Hmm, I'm still seeing an error for the check-in source: https://sourcemaps.io/report/1649196204940_https%3A%2F%2Fprod-va-gov-assets.s3-us-gov-west-1.amazonaws.com%2Fgenerated%2Fcheck-in.entry.js

I do see that 651.entry.js is served without gzip compression and check-in still uses it.

**Comment by acrollet** [comment]
Thanks for all your work on this btw!

**Comment by timwright12** [comment]
@acrollet where did you find the gzip information there? I know sentry can't read gzipped assets, so maybe that's what's going on

**Comment by timwright12** [comment]
Another sourcemap that validates https://sourcemaps.io/report/1649251552814_https%3A%2F%2Fwww.va.gov%2Fgenerated%2F1990s-edu-benefits.entry.js

**Comment by timwright12** [comment]
Ok, so... this one validates from va.gov, but not from the s3 bucket: 
https://prod-va-gov-assets.s3-us-gov-west-1.amazonaws.com/generated/1990s-edu-benefits.entry.js

**Comment by acrollet** [comment]
@timwright12 nice find! And sure enough check-in validates when using va.gov:

https://sourcemaps.io/report/1649257076558_https%3A%2F%2Fwww.va.gov%2Fgenerated%2Fcheck-in.entry.js

**Comment by acrollet** [comment]
It appears that the generated files are served without gzip compression from va.gov fwiw

**Comment by timwright12** [comment]
We found where the files are being gzipped: https://github.com/department-of-veterans-affairs/vets-website/blob/master/script/github-actions/deploy.sh

Not sure why the reverse proxy isn't respecting that. @acrollet how are you seeing the va.gov assets as non-gzipped? I just want to make sure we're seeing the same information

I don't think we want to remove gzip compression for Sentry to read files, but I just want to confirm that this is the issue. I also tried to upload sourcemaps/files directly into sentry, but it won't allow me to since it's behind the TIC/SOCKS
